qiskit/providers/basic_provider/basic_simulator.py

```diff
@@ -0,0 +1,744 @@
+
+        super().__init__(
+            provider=provider,
+            name="basic_simulator",
+            description="A python simulator for quantum experiments",
+            backend_version="0.1",
+            **fields,
+        )
+
+        self._target = target
+        self._configuration = None
+
+        # Internal simulator variables
+        self._local_random = np.random.RandomState()
+        self._classical_memory = 0
+        self._classical_register = 0
+        self._statevector = 0
+        self._number_of_cmembits = 0
+        self._number_of_qubits = 0
+        self._shots = 0
+        self._memory = False
+        self._initial_statevector = self.options.get("initial_statevector")
+        self._chop_threshold = self.options.get("chop_threashold")
+        self._qobj_config = None
+        self._sample_measure = False
+
+    @property
+    def max_circuits(self) -> None:
+        return None
+
+    @property
+    def target(self) -> Target:
+        if not self._target:
+            self._target = self._build_basic_target()
+        return self._target
+
+    def _build_basic_target(self) -> Target:
+        """Helper method that returns a minimal target with a basis gate set but
+        no coupling map, instruction properties or calibrations.
+
+        Returns:
+            The configured target.
+        """
+        # Set num_qubits to None to signal the transpiler not to
+        # resize the circuit to fit a specific (potentially too large)
+        # number of qubits. The number of qubits in the circuits given to the
+        # `run` method will determine the size of the simulated statevector.
+        target = Target(
+            description="Basic Target",
+            num_qubits=None,
+        )
+        basis_gates = [
+            "h",
+            "u",
+            "p",
+            "u1",
+            "u2",
+            "u3",
+            "rz",
+            "sx",
+            "x",
+            "cx",
+            "id",
+            "unitary",
+            "measure",
+            "delay",
+            "reset",
+        ]
+        inst_mapping = get_standard_gate_name_mapping()
+        for name in basis_gates:
+            if name in inst_mapping:
+                instruction = inst_mapping[name]
+            elif name == "unitary":
+                # This is a placeholder for an actual unitary gate
+                # to signal the transpiler not to decompose unitaries
+                # present in the circuit.
+                instruction = Gate(
+                    name=name,
+                    num_qubits=0,
+                    params=[Parameter("matrix")],
+                )
+            else:
+                raise BasicProviderError(
+                    "Gate is not a valid basis gate for this simulator: %s" % name
+                )
+
+            target.add_instruction(instruction, properties=None, name=name)
+        return target
+
+    def configuration(self) -> BackendConfiguration:
```

Comment:
Can you put a comment here to say this is a custom attribute on `BasicSimulator`? We discussed this is only needed for `assemble()`, but I am mildly worried people will use this as a model and expect `BackendConfiguration` to continue to exist.

Reply:
Also done in [8300a92](https://github.com/Qiskit/qiskit/pull/11583/commits/8300a929e2ae3abe90ef0d0060abdb39f95bd75e).